src/net_processing.cpp

```diff
@@ -1266,6 +1266,7 @@ bool static ProcessHeadersMessage(CNode *pfrom, CConnman *connman, const std::ve
                 LOCK(cs_main);
                 Misbehaving(pfrom->GetId(), nDoS);
             }
+            LOCK(cs_main);
```

Comment:
I think we can move this up to above the `if (nDoS > 0)` line, and then remove the `LOCK(cs_main)` that's inside that block?